lib/primitives/mtx.js

```diff
@@ -1824,8 +1881,19 @@ class CoinSelector {
 
   async select(coins) {
     this.init(coins);
-
-    if (this.hardFee !== -1) {
+    if (this.fill) {
+      // if sorting by value, reverse
+      // so smallest value is first
+      if (
+        this.selection === 'value' &&
```

Comment:
If we want to accept other sorting options, it would make more sense to add another sorting option that will order coins in reverse order in the first place. Because someone might want to add coins from "threshold" -> to lowest.

Reply:
Yeah, good call. I'll add `reverseValue` option for `selection` then and add a new sorting function

Reply:
I've implemented this now, but found it a little tricky with threshold because you only want to do threshold if you're sorting by reverseValue. So I added an extra assert that throws if threshold is set but `reverseValue` sorting is not.